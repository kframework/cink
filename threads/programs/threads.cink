int r; 
int f(int x) {
  return (r = x);
}
int main() {
  std::thread t1(f, 1);
  std::thread t1(f, 2);
  return r;
}
