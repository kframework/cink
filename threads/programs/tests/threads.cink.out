<T>
    <in>
        ListItem(#buffer ( "" ))
        ListItem(#istream ( 0 ))
    </in>
    <out>
        ListItem(#ostream ( 1 ))
        ListItem(#buffer ( "" ))
    </out>
    <threads>
        .Bag
    </threads>
    <store>
        #symScalLoc(1) |-> 2
        #symScalLoc(2) |-> lambda ( (int x) @ (return (r = x) ;) )
        #symScalLoc(3) |-> lambda ( .Decls @ (((std::thread t1 ( f , 1 ) ;) (
          std::thread t1 ( f , 2 ) ;)) (return r ;)) )
        #symScalLoc(4) |-> 1
        #symScalLoc(5) |-> 2
    </store>
    <genv>
        f |-> #symScalLoc(2)
        main |-> #symScalLoc(3)
        r |-> #symScalLoc(1)
    </genv>
</T>
